### PR TITLE
Add `CITATION.cff` file

### PR DESCRIPTION
**Pull request by cgobat**
## PR summary

Two years ago, in a9d5a89, the [repository's BibTeX citation file](../blob/main/CITATION.bib) was moved and renamed to be compatible with GitHub's new citation format. Building upon that, this PR adds a `CITATION.cff` file: [CFF (**c**itation **f**ile **f**ormat)](https://citation-file-format.github.io/) files are special YAML files with native GitHub support and direct integration with the ["Cite this repository" feature](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files). CFF is capable of providing incredibly rich metadata, and isn't LaTeX- or language-specific, so CFF files can easily be exported in a *variety* of popular citation formats.

As an example of such a file in use, see [`CITATION.cff` in pandas-dev/pandas](https://github.com/pandas-dev/pandas/blob/main/CITATION.cff) (though the one added in this PR contains even more detail!).

~~Because the CFF definition uses [`jsonschema`](https://json-schema.org/), CFF files can be easily validated (e.g. using [`cffconvert`](https://github.com/citation-file-format/cff-converter-python), or through GitHub Actions, using [dieghernan/cff-validator](https://github.com/dieghernan/cff-validator)). A workflow to do this automatically when the file is altered has also been added, in [`.github/workflows/cff-validator.yml`](https://github.com/cgobat/mpl-contrib/blob/add-cff/.github/workflows/cff-validator.yml).~~

This may or may not *replace* the existing [`CITATION.bib`](../blob/main/CITATION.bib)&mdash;some further discussion about the role of that file is warranted prior to removal.

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html) (validated using [`cffconvert`](https://github.com/citation-file-format/cff-converter-python))
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

File: .github/workflows/cff-validator.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - CITATION.cff
+  workflow_dispatch:
+
+name: CITATION.cff
+jobs:
+  Validate-CITATION-cff:
+    runs-on: ubuntu-latest
+    name: Validate CITATION.cff
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tacaswell) [inline .github/workflows/cff-validator.yml:13]: Why does this need a secret?

**Comment by tacaswell** [comment]
I am 👍🏻 on adding this file, but 👎🏻 on adding the validation workflow.  This file is going to be effectively static so I do not anticipate there being a need re-validate that we have not messed the file up.

File: .github/workflows/cff-validator.yml (hunk 1 above)
Comment (cgobat) [reply .github/workflows/cff-validator.yml:13]: Maybe I don't understand how workflows work&mdash;isn't a key/token always required in order to run/access the environment?

Moot point in any case if we're not going to add this workflow. 👍

**Comment by cgobat** [comment]
Thanks @tacaswell. I will remove the `cff-validator.yml` file from the workflows then.

**Comment by tacaswell** [comment]
@cgobat Are you comfortable squashing this to one commit?

**Comment by cgobat** [comment]
Yep&mdash;however you want to do it is fine by me.

**Comment by tacaswell** [comment]
Can you please squash it then :)

**Comment by cgobat** [comment]
Sure, done. Wasn't sure whether you were going to do it on your side as part of the merge operation.

**Review by tacaswell — APPROVED** [review]

**Pull request merged**